client/app/dbaas/logs/detail/options/logs-options.service.js

```diff
@@ -0,0 +1,136 @@
+class LogsOptionsService {
+    constructor ($translate, $window, OvhApiOrderCartServiceOption, ServiceHelper, OvhApiDbaasLogs, LogOptionConstant) {
+        this.OvhApiOrderCartServiceOption = OvhApiOrderCartServiceOption;
+        this.ServiceHelper = ServiceHelper;
+        this.$translate = $translate;
+        this.$window = $window;
+        this.OvhApiDbaasLogs = OvhApiDbaasLogs;
+        this.LogOptionConstant = LogOptionConstant;
+    }
+
+    /**
+     * returns the transformed option. Meant to be used for the available options
+     *
+     * @param {any} option
+     * @returns the transformed option
+     * @memberof LogsOptionsService
+     */
+    transformOption (option) {
+        option.quantity = 0;
+        option.price = option.prices[0].price.value;
+        option.priceText = option.prices[0].price.text;
+        option.type = this.$translate.instant(`${option.planCode}-type`);
+        option.detail = this.$translate.instant(`${option.planCode}-detail`);
+    }
+
+    /**
+     * returns the list of options available for selection
+     *
+     * @param {any} serviceName
+     * @returns promise which will be resolve to an array of options objects
+     * @memberof LogsOptionsService
+     */
+    getOptions (serviceName) {
+        const transformOption = this.transformOption.bind(this);
+        return this.OvhApiOrderCartServiceOption.Lexi().get({
+            productName: this.LogOptionConstant.productName,
+            serviceName
+        }).$promise
+            .then(response => {
+                _.map(response, option => transformOption(option));
```

Comment:
For readability's sake 

_.map(response, option => transformOption(option)); looks weird since you don't use the return value of _map.

I'd use _.forEach instead since transformOption changes the value contained in option reference :)

Reply:
That's right. I have done this change :)